browser/extensions/api/gemini_api.cc

```diff
@@ -0,0 +1,277 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/extensions/api/gemini_api.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/environment.h"
+#include "brave/browser/profiles/profile_util.h"
+
+#include "brave/common/extensions/api/gemini.h"
+#include "brave/common/extensions/extension_constants.h"
+#include "brave/common/pref_names.h"
+#include "brave/browser/gemini/gemini_service_factory.h"
+#include "brave/components/gemini/browser/gemini_service.h"
+#include "chrome/browser/extensions/api/tabs/tabs_constants.h"
+#include "chrome/browser/extensions/extension_tab_util.h"
+#include "chrome/browser/infobars/infobar_service.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/prefs/pref_service.h"
+#include "extensions/browser/extension_util.h"
+#include "components/country_codes/country_codes.h"
+#include "brave/browser/gemini/gemini_util.h"
+
+namespace {
+
+GeminiService* GetGeminiService(content::BrowserContext* context) {
+  return GeminiServiceFactory::GetInstance()
+      ->GetForProfile(Profile::FromBrowserContext(context));
+}
+
+bool IsGeminiAPIAvailable(content::BrowserContext* context) {
```

Comment:
this shouldn't be handled here, it should be handled in https://github.com/brave/brave-core/pull/6062/files#diff-1892c3947f1038538c42dd52ce761684R19 by returning null for these conditions. Otherwise we're incorrectly starting the service for profiles that shouldn't have it